QHyper/solvers/quantum_annealing/advantage.py

```diff
@@ -19,15 +37,69 @@ class Advantage(Solver):
         The region in which D-Wave Advantage is available.
     """
 
-    def __init__(self, problem: Problem, region: str = "eu-central-1") -> None:
+    def __init__(self, problem: Problem, region: str = "eu-central-1",
+                 optimizer: Optimizer | None = None) -> None:
         self.problem = problem
         self.region = region
+        self.optimizer = optimizer
 
     def solve(self, params_inits: dict[str, Any] = {}) -> Any:
-        sampler = DWaveSampler(
-            region=self.region, solver='Advantage_system5.4')
-        qubo = Converter.create_qubo(
-            self.problem, params_inits.get("weights", []))
-        sampleset = EmbeddingComposite(sampler).sample_qubo(qubo.terms)
+        sampler = DWaveSampler(solver='Advantage_system4.1')
+        embedding_compose = EmbeddingComposite(sampler)
+
+        opt_result = None
+        if self.optimizer:
+            args = (
+                np.array(params_inits["weights"]).flatten() if params_inits["weights"]
+                else None
+            )
+
+            opt_wrapper = OptimizerFunction(embedding_compose, self.problem)
+            result = self.optimizer.minimize(opt_wrapper, args)
+            opt_result = result.params
+
+        qubo_arguments = opt_result if self.optimizer else params_inits.get("weights", [])
+        sampleset = run_advantage(self.problem, embedding_compose, qubo_arguments)
+        result = self.prepare_solver_result(sampleset.first.sample, qubo_arguments)
+
+        return result
```

Comment:
return results similar to CQM and DQM 
https://github.com/qc-lab/QHyper/blob/kacper3615/dqm_cqm_fix/QHyper/solvers/quantum_annealing/cqm.py#L59